### PR TITLE
all added backend changed for treeViz on flow

### PR DESCRIPTION
**Pull request by msrimat**
This change is [Reviewable](https://reviewable.io/reviews/h2oai/h2o-3/91)

### CODE REVIEW

**Comment by abal5** [comment]
Can one of the admins verify this patch?

**Comment by arnocandel** [comment]
Looks great. Can we add some tests to ensure that the POJO tree and the JSON tree have the same structure?

File: h2o-py/h2o/__init__.py [hunk 1]
```diff
@@ -23,7 +23,7 @@
 # one would have to import it from h2o.frames.
 from h2o.frame import H2OFrame  # NOQA
 
-__version__ = "SUBST_PROJECT_VERSION"
+__version__ = "3.11.0.99999"
```
Comment (ivyywang) [inline h2o-py/h2o/__init__.py]: should we hard code this?

**Comment by st-pasha** [comment]
Is "Copy of DRF_Example" file really needed?

File: h2o-py/h2o/__init__.py (hunk 1 above)
Comment (st-pasha) [inline h2o-py/h2o/__init__.py]: This seems like a side-effect from running a build script. Should probably be reverted

File: h2o-algos/src/main/java/hex/api/TreeVizHandler.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import water.*;
+import water.api.Handler;
+
+//Created by Manu Srimat
+
+public class TreeVizHandler extends Handler {
+
+  public TreeVizSchemaV3 get_trees(int version, TreeVizSchemaV3 args){
+    // get the model by key, fetch the trees set them to schema
+    GBMModel m = DKV.getGet(args.modelKey.key());
```
Comment (st-pasha) [inline h2o-algos/src/main/java/hex/api/TreeVizHandler.java]: No arguments verification? The client can send anything...

**Comment by st-pasha** [comment]
Also a bunch of temporary files included in this diff: testerfile, testerfile.rtf, Untitled Flow, tojson.tar.gz, tmp/drf_2a66d448_10e5_49b4_b5db_73cd746b50ea.java, tmp/h2o-genmodel.jar, tmpdir/drf_d3bcdacb_429a_43f5_bb32_7274d79721ae.java, tmpdir/h2o-genmodel.jar

File: h2o-py/h2o/__init__.py (hunk 1 above)
Comment (mmalohlava) [inline h2o-py/h2o/__init__.py]: has to be reverted

**Comment by mmalohlava** [comment]
Great prototype solution, but needs an h2o engineer to clean up. My comments:

  - using Gson to generate a simple JSON looks too expensive (especially for  - larger ensembles) - we should keep memory footprint low. Using `SB` should be good enough.
  - missing tests - this is crucial - we need to have at least jUnits ensuring the isomorphism between in-memory tree and generated JSON. Typical cases to tests are
     - a single node tree
     - 3nodes tree
     - full binary tree
     - unbalanced tree with one  long path
     - ...
  - it would be nice to have some kind of "tree-pagination" - e.g., the REST API could provide only a subset of trees, or the REST API could accept a node id to generate only a subtree from a selected tree

**Pull request closed without merging**